src/mainboard/protectli/vault_jsl/mainboard.c

```diff
@@ -108,6 +111,150 @@ static void descriptor_patch_pcie8_lane(int is_nvme)
+		return 0;
+
+	eth = pcidev_path_behind(parent->link_list, PCI_DEVFN(0, 0));
+	if (!eth || !eth->enabled)
+		return 0;
+
+	/* Only i226 Ehternet Controller */
+	if (eth->vendor != PCI_VID_INTEL || eth->device != 0x125c)
+		return 0;
+
+	return smbios_write_type41(
+		current, handle,
+		"Onboard - Ethernet",		/* name */
+		instance++,			/* instance */
+		0,				/* segment */
+		eth->bus->secondary,		/* bus */
+		PCI_SLOT(eth->path.pci.devfn),	/* device */
+		PCI_FUNC(eth->path.pci.devfn),	/* function */
+		SMBIOS_DEVICE_TYPE_ETHERNET);	/* device type */
+}
+
+static int smbios_type41_write_ethernet_behind_switch(struct device *root_port, int *handle,
+						      unsigned long *current)
+{
+	struct device *switch_dev;
+	struct device *child;
+	int len = 0;
+
+	if (!root_port || !root_port->enabled)
+		return 0;
+
+	switch_dev = pcidev_path_behind(root_port->link_list, PCI_DEVFN(0, 0));
+	if (!switch_dev || !switch_dev->enabled)
+		return 0;
+
+	/* Only ASMedia switch */
+	if (switch_dev->vendor != 0x1b21 || switch_dev->device != 0x2806)
+		return 0;
+
+	/* After upstream port there is also one downstream port */
+	switch_dev = pcidev_path_behind(switch_dev->link_list, PCI_DEVFN(0, 0));
+	if (!switch_dev || !switch_dev->enabled)
+		return 0;
+
+	/* Only ASMedia switch */
+	if (switch_dev->vendor != 0x1b21 || switch_dev->device != 0x2806)
+		return 0;
+
+	/* Loop through all downstream ports of the switch */
+	for (child = switch_dev->bus->children; child; child = child->sibling) {
+		printk(BIOS_DEBUG, "Switch DS port %s [%04x:%04x]\n",
+			dev_path(child), child->vendor, child->device);
+		len += smbios_type41_write_ethernet(child, handle, current);
+	}
+
+	return len;
+}
+
+static int mainboard_smbios_data(struct device *dev, int *handle,
+				 unsigned long *current)
+{
+	int len = 0;
+
+	len += smbios_write_type41(
+		current, handle,
+		"Onboard - USB Controller",	/* name */
+		0,				/* instance */
+		0,				/* segment */
+		0,				/* bus */
+		PCH_DEV_SLOT_XHCI,		/* device */
+		0,				/* function */
+		SMBIOS_DEVICE_TYPE_OTHER);	/* device type */
+
+	len += smbios_write_type41(
+		current, handle,
+		"Onboard - HECI",		/* name */
+		0,				/* instance */
+		0,				/* segment */
+		0,				/* bus */
+		PCH_DEV_SLOT_CSE,		/* device */
+		0,				/* function */
+		SMBIOS_DEVICE_TYPE_OTHER);	/* device type */
+
+	len += smbios_write_type41(
+		current, handle,
+		"Onboard - eMMC",		/* name */
+		0,				/* instance */
+		0,				/* segment */
+		0,				/* bus */
+		PCH_DEV_SLOT_STORAGE,		/* device */
+		0,				/* function */
+		SMBIOS_DEVICE_TYPE_OTHER);	/* device type */
+
+	if (!CONFIG(BOARD_PROTECTLI_V1610)) {
+		len += smbios_type41_write_ethernet(PCH_DEV_PCIE1, handle, current);
+		len += smbios_type41_write_ethernet(PCH_DEV_PCIE2, handle, current);
+	} else {
+		len += smbios_type41_write_ethernet_behind_switch(PCH_DEV_PCIE1, handle,
+								  current);
+		len += smbios_type41_write_ethernet(PCH_DEV_PCIE5, handle, current);
+	}
```

Comment:
Not sure if this part is correct for all models.  https://github.com/Dasharo/coreboot/blob/c7aa69f948e264b6217318f29059e6f0f1cd3395/src/mainboard/protectli/vault_jsl/override_v12xx.cb has only 2 LANs which added below this block.

Maybe this was the intention:
```suggestion
	if (CONFIG(BOARD_PROTECTLI_V1410)) {
		len += smbios_type41_write_ethernet(PCH_DEV_PCIE1, handle, current);
		len += smbios_type41_write_ethernet(PCH_DEV_PCIE2, handle, current);
	} else if (CONFIG(BOARD_PROTECTLI_V1610)) {
		len += smbios_type41_write_ethernet_behind_switch(PCH_DEV_PCIE1, handle,
								  current);
		len += smbios_type41_write_ethernet(PCH_DEV_PCIE5, handle, current);
	}
```

However, https://github.com/Dasharo/coreboot/blob/c7aa69f948e264b6217318f29059e6f0f1cd3395/src/mainboard/protectli/vault_jsl/override_v1610.cb has WiFi on `PCH_DEV_PCIE5` as referenced in https://github.com/Dasharo/coreboot/pull/708/commits/dbd4f535fb7d81c3f7a14939e708932035642480 so the second line in the second `if` seems to not be needed.  Checks for device & vendor numbers in the functions above likely make this code work as is.

Reply:
Applied in https://github.com/Dasharo/coreboot/pull/708/commits/aaab86af74f73a868b0345c57afdba9204fdd56d